var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -835,3 +860,14 @@ def setup_run_environment(self, env):
         if "+python" in self.spec:
             env.prepend_path("LD_LIBRARY_PATH", self.prefix.lib)
             env.prepend_path("LD_LIBRARY_PATH", self.prefix.lib64)
+
+    @run_after("install")
+    def install_python(self):
+        # do LAMMPS Python package installation using pip
+        if self.spec.satisfies("@20230328: +python"):
```

Comment:
So the Python variant exists for all versions but you only run this for new versions. Was this automatically installed in the past or something?

Reply:
In the past it was part of the `install` target, but that was removed in favor of using pip directly. 

For the majority of users who build LAMMPS directly via CMake or Make, there is a `install-python` target that installs the Python package along with the LAMMMPS shared library in the user's HOME. However, that mechanism isn't meant first packaging. Instead it should be treated as a regular Python package and installed via pip.